nationalarchives/ds-wagtail
# CHORE: Add custom image model

## About these changes

Wagtail has supported custom image models since forever - they are perfect for projects like this, where you gradually discover more and more things you want to do with images, and want additional fields to accommodate that. 

By adding these optional custom fields here, we actually negate the need for the `Highlight` snippet model (we can just add a "Highlights" collection under "Images", and add them as images). We can also store transcriptions/translations directly on the image model, negating the need to capture that data for individual gallery images on the `RecordArticlePage` type (and possibly other places in future)

## How to check these changes

Try not to get too hung up on labels/choices/help text etc... we don't have all of that absolutely figured and specced yet, but I've been through all of the changes with Lauren via screenshare, and we're happy to go with the current configuration and tweak later as required. There are some things you can check from a dev perspective though:

- Check that images retain the same ID before and after migrating
- Check a page that used images BEFORE migrating retains references to those same images in the new table
- Check that you can add a new image after migrating

## Before assigning to reviewer, please make sure you have
